### PR TITLE
backport kselftest and ltp from lkft

### PR DESCRIPTION
**Pull request by bobolmw**

### CODE REVIEW

File: tests/runall.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+[ ! -f /opt/kselftests/run_kselftest.sh ] && echo "Please install kselftests to /opt/kselftests first" && exit 1
+[ ! -f /opt/ltp/runltp ] && echo "Please install ltp to /opt/ltp first" && exit 1
+
```
Comment (adamliyi) [inline tests/runall.sh:5]: Could you please add script or document on how to build and install ltp and kselftests? We'd like to make the test as automatic as possible.

**Review by adamliyi — CHANGES_REQUESTED** [review]

File: tests/runall.sh (hunk 1 above)
Comment (bobolmw) [reply tests/runall.sh:5]: Done

File: tests/lkft/ltp/skipfile-lkft.yaml [hunk 2]
```diff
@@ -0,0 +1,334 @@
+# This skipfile is consumable using skipgen
```
Comment (adamliyi) [inline tests/lkft/ltp/skipfile-lkft.yaml:1]: Are there any ltp test cases need to skip on Altra?

File: tests/lkft/kselftest/skipfile-lkft.yaml [hunk 3]
```diff
@@ -0,0 +1,391 @@
+# This skipfile is consumable using skipgen
```
Comment (adamliyi) [inline tests/lkft/kselftest/skipfile-lkft.yaml:1]: Are there any kselftest test cases we need to skip on Altra? If yes, could you add comments to 'reason' field?

File: tests/lkft/kselftest/skipfile-lkft.yaml [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+      - hi6220-hikey
+      - juno-r2
+      - qemu_arm64
+      - x15
+      - nxp-ls2088
+    branches: all
+    tests:
+      - breakpoints:breakpoint_test_arm64
+
+  - reason:
```
Comment (adamliyi) [inline tests/lkft/kselftest/skipfile-lkft.yaml:75]: Is this case only for Altra?

File: tests/lkft/kselftest/skipfile-lkft.yaml [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+    url:
+    environments: all
+    boards: all
+    branches: all
+    tests:
+      - breakpoints:step_after_suspend_test
+
+  - reason:
```
Comment (adamliyi) [inline tests/lkft/kselftest/skipfile-lkft.yaml:83]: Only for Altra?

**Review by adamliyi — CHANGES_REQUESTED** [review]
Thanks. I run the scripts. It works well. 

One comment:
Is it OK to move bin/, lib/, patches/ to lkft, e.g:
```
lkft/bin/
lkft/lib/
lkft/patches/
```
We are going to add directory for kernel patch unit test, e.g: /tests/kernel_patch_test/ . But if you think other test suite will also use bin/, lib/, patches/, it is OK to keep it in /tests/ .

**Comment by bobolmw** [comment]
> Thanks. I run the scripts. It works well.
> 
> One comment:
> Is it OK to move bin/, lib/, patches/ to lkft, e.g:
> 
> ```
> lkft/bin/
> lkft/lib/
> lkft/patches/
> ```
> 
> We are going to add directory for kernel patch unit test, e.g: /tests/kernel_patch_test/ . But if you think other test suite will also use bin/, lib/, patches/, it is OK to keep it in /tests/ .

It's ok to keep this dir structure, since lkft testcase script have some relative path depends on it. And it's good that let our script to reference lib/sh-test-lib script.

For comments on skipfiles:
         The skipfiles in this patches is copied from lkft directly. And I think it's better to edit those skipfiles during testcase fails fixing stage.

**Pull request merged**